drivers/serial/uart_stm32.c

```diff
@@ -314,6 +293,15 @@ static int uart_stm32_init(struct device *dev)
 				 LL_USART_PARITY_NONE,
 				 LL_USART_STOPBITS_1);
 
+	/* HW flow control */
+	if (config->rts_cts) {
```

Comment:
You should provide a start up configuration otherwise behavior is undefined here.